relay-server/src/utils/rate_limits.rs

```diff
@@ -403,6 +403,67 @@ impl Enforcement {
             })
     }
 
+    /// Applies the [`Enforcement`] on the [`Envelope`] by removing all items that were rate limited
+    /// and emits outcomes for each rate limited category.
+    pub fn apply_with_outcomes(
```

Comment:
We should make `track_outcomes` private to make sure that all users call `apply_with_outcomes` instead.

Reply:
I was thinking about it, I will have to adjust the code in the processor rate limit checks.

Reply:
I totally missed the other callsite because I had the cfg flag disabled.